flash/data/data_module.py

```diff
@@ -164,25 +167,35 @@ def _show_batch(self, stage: RunningStage, reset: bool = True) -> None:
                 iter_dataloader = self._reset_iterator(stage)
                 _ = next(iter_dataloader)
             data_fetcher: BaseVisualization = self.data_fetcher
+            # list of functions to visualise
+            data_fetcher._fcn_white_list = self._fcn_white_list
             data_fetcher._show(stage)
             if reset:
                 self.viz.batches[stage] = {}
 
-    def show_train_batch(self, reset: bool = True) -> None:
+    def show_train_batch(self, name: str = 'load_sample', reset: bool = True) -> None:
         """This function is used to visualize a batch from the train dataloader."""
-        self._show_batch(_STAGES_PREFIX[RunningStage.TRAINING], reset=reset)
+        stage_name: str = _STAGES_PREFIX[RunningStage.TRAINING]
```

Comment:
Move this logic to `_show_batch` to reduce duplicated code and raise a MisConfigurationError is the provided names aren't in _Preprocess_funcs.

Reply:
done